activerecord/test/cases/locking_test.rb

```diff
@@ -821,6 +821,20 @@ def test_no_locks_no_wait
       assert first.end > second.end
     end
 
+    def test_lock_with_true_argument_deprecation
+      person = Person.find 1
+      assert_deprecated(ActiveRecord.deprecator) { person.lock!(true) }
+      assert_not_deprecated(ActiveRecord.deprecator) { person.lock!("FOR UPDATE") }
+      assert_not_deprecated(ActiveRecord.deprecator) { person.lock!(Arel.sql("FOR UPDATE")) }
+    end
+
+    def test_with_lock_with_true_argument_deprecation
+      person = Person.find 1
+      assert_deprecated(ActiveRecord.deprecator) { person.with_lock(true) { } }
+      assert_not_deprecated(ActiveRecord.deprecator) { person.with_lock("FOR UPDATE") { } }
+      assert_not_deprecated(ActiveRecord.deprecator) { person.with_lock(Arel.sql("FOR UPDATE")) { } }
+    end
```

Comment:
There's no need to test for `false` or `nil` here unlike with the relation `#lock` since the method is only called with truthy values, which makes sense cause calling `record.lock!(false)` would be very unusual...

Ref: https://github.com/rails/rails/blob/276b655f3606cf81b26464d135db9f24bacc1f0f/activerecord/lib/active_record/persistence.rb#L838-L839